activesupport/lib/active_support/cache/serializer_with_fallback.rb

```diff
@@ -40,6 +46,40 @@ def load(dumped)
       end
 
       private
+        BARE_STRING_SIGNATURES = {
+          255 => Encoding::UTF_8,
+          254 => Encoding::BINARY,
+          253 => Encoding::US_ASCII,
+        }
+        BARE_STRING_TEMPLATE = "CEl<"
+        BARE_STRING_VERSION_INDEX = [0, 0.0, 0].pack(BARE_STRING_TEMPLATE).bytesize
+
+        def try_dump_bare_string(entry)
+          if entry.value.instance_of?(String) && signature = BARE_STRING_SIGNATURES.key(entry.value.encoding)
+            packed = [
+              signature,
+              entry.expires_at || -1.0,
+              entry.version&.bytesize || -1,
+            ].pack(BARE_STRING_TEMPLATE)
+
+            packed << entry.version if entry.version
+            packed << entry.value
+          end
+        end
+
+        def try_load_bare_string(dumped)
+          if encoding = BARE_STRING_SIGNATURES[dumped.getbyte(0)]
+            _, expires_at, version_length = dumped.unpack(BARE_STRING_TEMPLATE)
+            value_index = BARE_STRING_VERSION_INDEX + [version_length, 0].max
+
+            Cache::Entry.new(
+              dumped.byteslice(value_index..-1).force_encoding(encoding),
+              version: dumped.byteslice(BARE_STRING_VERSION_INDEX, version_length),
```

Comment:
Should we assume the version is UTF-8 encoded?

Reply:
I was wondering about that.  `FlatCacheEntryCoder` leaves it as `Encoding::BINARY`, right?  Or is it converted prior to `FlatCacheEntryCoder#load`?

I think that `version` will be an ASCII-only UTF-8-encoded string in the vast majority of cases, and either encoding would work.  But to be sure, I have added a `force_encoding(Encoding::UTF_8)` in `load`, as well as a `version.encoding` check in `dump` so that we only deal with UTF-8 `version` strings.

Reply:
> FlatCacheEntryCoder leaves it as Encoding::BINARY, right? 

Yes.

> so that we only deal with UTF-8 version strings.

That seems like the proper thing to do IMO.